### DISCUSSION TITLE
"Edit Database" dialog incorrectly limits submitting valid connection information.

### DISCUSSION THREAD

**Discussion by jasonburrows** [apache/superset#18506] [Ideas]
### Expected results

I expect that when I provide a 'host' param with a unix socket in the "Additional Parameters" field, it will allow me to submit the form without requiring the "Host" and "Port" fields. 

### Actual results

It forces the user to enter Host and Port, even though they are not used for this connection style, and tests against them to ensure they will accept connections (which they will not).

#### Screenshots

![Screen Shot 2021-09-16 at 10 34 32 AM](https://user-images.githubusercontent.com/18748006/133658859-f8056f83-7d9e-44de-9247-b1a3871c4a2a.png)

#### How to reproduce the bug

1. Go to "+ Database"
2. Enter valid unix socket information in the "Additional Parameters" field.
3. Watch with frustration as the form blocks submission due to lack of IP/Port which will go unused anyway.

### Environment

(All versions are what are included in your posted apache/superset docker image)

- superset version: 1.3.1rc1
- python version: 3.7.9
- node.js version: 14
- any feature flags active:  Default for published docker file

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ X] I have reproduced the issue with at least the latest released version of superset.
- [X ] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I have Superset running in Google Cloud Run and it is using a Google Cloud SQL Postgres DB for metadata.  This works fine as I can configure the connection string for the metadata DB directly.  Due to GCP limitations, it is recommended to use the unix socket as a connection string rather than host/port.

I want to target DBs for analytics that are ON THE SAME HOST it is already connected to for metadata (so clearly it is capable of connecting to that host).  I am able to enter the unix socket information in the "Additional Parameters" section of the Edit Database dialog box, but because it requires the Host and Port, and tests against that Host and Port, it won't let me submit what is otherwise valid connection information.

It seems like it is purely a UI limitation and that it is blocking the submission of otherwise valid connection information.

**Comment by junlincc** [comment]
@jasonburrows it seems to be an easy tweak. would you like to open a PR to unblock yourself and those who encountered same challenge? 

cc @betodealmeida
